server/aesdsocket.c

```diff
@@ -0,0 +1,477 @@
+    if(close(file_fd) == -1)
+    {
+        //Else, error occurred, print it to syslog and finish program
+        syslog(LOG_ERR, "Could not close log file: %s", strerror(errno));
+        exit(1);
+    }
+    //Delete the file
+    if(remove(LOG_PATH) == -1)
+    {
+        //Else, error occurred, print it to syslog and finish program
+        syslog(LOG_ERR, "Could not remove log file: %s", strerror(errno));
+        exit(1);
+    }
+}
+
+/**
+* socketserver
+* @brief Creates a server listening to port 9000
+* 
+* The server functionality is described in more detail at
+* the file header.
+*
+* @return void
+*/
+void socketserver(int sck)
+{
+    //Start listening to addr+port
+    if(listen(sck, SERVER_QUEUE) == -1)
+    {
+        syslog(LOG_ERR, "An error occurred listening the socket: %s", strerror(errno));
+        exit(1);        
+    }
+
+    syslog(LOG_INFO, "The server is listening to port 9000");
+
+    //Create the file that will log the messages received
+    int file_fd = open(LOG_PATH, O_CREAT | O_RDWR | O_TRUNC, 0666);
+    if(file_fd == -1)
+    {
+        syslog(LOG_ERR, "Could not create the log file: %s", strerror(errno));
+        exit(1);  
+    }
+
+    //Start a loop of receiving contents
+    int file_size = 0;
+    char *recv_data = NULL;
+
+    while(!graceful_exit)
+    {
+        struct sockaddr_storage client_addr;
+        socklen_t addr_size = sizeof client_addr;
+
+        //Accept a connection
+        int connection_fd = accept(sck, (struct sockaddr *) &client_addr, &addr_size);
+        if(connection_fd == -1)
+        {
+            if(errno == EINTR)
+            {
+                //The signal has set "graceful_exit" and the next while iteration will not happen
+                break;
+            }
+            else
+            {
+                syslog(LOG_ERR, "An error occurred accepting a new connection to the socket: %s", strerror(errno));
+                exit(1);
+            }
+        }
+
+        print_accepted_conn(client_addr);
+
+        //Wait for data
+        int recv_ret;
+        int index = 0;
+        //Keep track of how many RECV_BUFF_LEN chunks "recv_data" has
+        int chunks = 1;
+        
+        //Start with a size, potentially increasing it if an entire packet cannot fit into it
+        recv_data = malloc(sizeof(char)*RECV_BUFF_LEN*chunks);
+        if(!recv_data)
+        {
+            syslog(LOG_ERR, "Could not malloc: %s", strerror(errno));
+            exit(1);  
+        }
+
+        do
+        {
+            recv_ret = recv(connection_fd, &recv_data[index], RECV_BUFF_LEN, 0);
+            if(recv_ret == -1)
+            {
+                syslog(LOG_ERR, "An error occurred reading from the socket: %s", strerror(errno));
+                exit(1);  
+            }
+            index += recv_ret;
+
+            if(index != 0)
+            {
+                //Check if the last value received is "\n"
+                if(recv_data[index - 1] == '\n')
```

Comment:
@jordicrosmompart what would happen to the logic here if the `recv` buffer contained two packets?